<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>enhancerHomologSearch Guide • enhancerHomologSearch</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="enhancerHomologSearch Guide">
<meta property="og:description" content="enhancerHomologSearch">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">enhancerHomologSearch</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/enhancerHomologSearch.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jianhong/enhancerHomologSearch/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="enhancerHomologSearch_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>enhancerHomologSearch Guide</h1>
                        <h4 class="author">Jianhong Ou</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jianhong/enhancerHomologSearch/blob/master/vignettes/enhancerHomologSearch.Rmd"><code>vignettes/enhancerHomologSearch.Rmd</code></a></small>
      <div class="hidden name"><code>enhancerHomologSearch.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>There is an increasing requirement for the tools to identify of putative mammalian orthologs to enhancers in species other than human and mouse, such as zebrafish, which is lacking whole genome comparison analysis data. Take zebrafish as an example, there are two major methods to identify the orthologs to enhancers in human and mouse,</p>
<ol style="list-style-type: decimal">
<li><p>use the whole genome comparison analysis data and conservation data<span class="citation"><sup>1</sup></span>,</p></li>
<li><p>use spotted gar genome as bridge genome to search the orthologs<span class="citation"><sup>2</sup></span>.</p></li>
</ol>
<p>Both methods will work well in the coding region. However, there is lacking comparative data in distal regulation region such as enhancers and silencers.</p>
<p>In 2020, Emily S. Wong et. al. provides a new method for identification of putative human orthologs to enhancers of zebrafish<span class="citation"><sup>3</sup></span>. They used the method to interrogate conserved syntenic regions and human and mouse using candidate sponge enhancer sequences. First, they looked for overlap with available functional genomics information. For example, they used mouse ENCODE data to infer enhancer activity based on histone marks in specific tissues. Second, they select the best-aligned region by whole genome alignment from the candidates regions for human and mouse as orthologs. This method provides the possibility to search orthologs for enhancers or silencers even there is not genome comparative data available.</p>
<p>This package is modified from Wong’s methods and provide the easy-to-use script for researchers to quick search putative mammalian orthologs to enhancers. The modified algorithm is: The candidate regions were determined by ENCODE histone marks (default is H3K4me1) in specific tissue for human and mouse. The mapping score were calculated by pairwise alignment between enhancer sequences and candidates by global sequence alignment<span class="citation"><sup>4</sup></span>. The Z-score were calculated from mapping score and then converted to P-value based on two-side test from a normal distribution. The candidates were filtered by p-value and distance from the TSS of target homologs. And then the top candidates from human and mouse were aligned to each other and exported as multiple alignments with given enhancer.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>First install <code>enhancerHomologSearch</code> and other packages required to run the examples. Please note the example dataset used here is from zebrafish. To run analysis with dataset from a different species or different assembly, please install the corresponding Bsgenome and TxDb. For example, to analyze cattle data aligned to bosTau9, please install BSgenome.Btaurus.UCSC.bosTau9, and TxDb.Btaurus.UCSC.bosTau9.refGene. You can also generate a TxDb object by functions makeTxDbFromGFF from a local gff file, or makeTxDbFromUCSC, makeTxDbFromBiomart, and makeTxDbFromEnsembl, from online resources in GenomicFeatures package.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"BiocManager"</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BiocManager</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"enhancerHomologSearch"</span>,
                       <span class="st">"BiocParallel"</span>,
                       <span class="st">"BSgenome.Drerio.UCSC.danRer10"</span>,
                       <span class="st">"BSgenome.Hsapiens.UCSC.hg38"</span>,
                       <span class="st">"BSgenome.Mmusculus.UCSC.mm10"</span>,
                       <span class="st">"TxDb.Hsapiens.UCSC.hg38.knownGene"</span>,
                       <span class="st">"TxDb.Mmusculus.UCSC.mm10.knownGene"</span>,
                       <span class="st">"org.Hs.eg.db"</span>,
                       <span class="st">"org.Mm.eg.db"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>If you have trouble in install ribosomeProfilingQC, please check your R version first. The <code>enhancerHomologSearch</code> package require R &gt;= 4.1.0.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">R.version</span></code></pre></div>
<pre><code>##                _                           
## platform       x86_64-pc-linux-gnu         
## arch           x86_64                      
## os             linux-gnu                   
## system         x86_64, linux-gnu           
## status                                     
## major          4                           
## minor          1.1                         
## year           2021                        
## month          08                          
## day            10                          
## svn rev        80725                       
## language       R                           
## version.string R version 4.1.1 (2021-08-10)
## nickname       Kick Things</code></pre>
</div>
<div id="step-1-prepare-target-enhancer-sequences-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-1-prepare-target-enhancer-sequences-" class="anchor"></a>Step 1, prepare target enhancer sequences.</h2>
<p>In this example, we will use an enhancer of <code>lepb</code> gene in zebrafish.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load genome sequences</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Drerio.UCSC.danRer10</span><span class="op">)</span>
<span class="co"># define the enhancer genomic coordinates</span>
<span class="va">LEN</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"chr4"</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="fl">19050041</span>, <span class="fl">19051709</span><span class="op">)</span><span class="op">)</span>
<span class="co"># extract the sequences as Biostrings::DNAStringSet object</span>
<span class="op">(</span><span class="va">seqEN</span> <span class="op">&lt;-</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">BSgenome.Drerio.UCSC.danRer10</span>, <span class="va">LEN</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 1:
##     width seq
## [1]  1669 TGGCATACACAGCAAACATCATGAATTTAATTTA...TAGATAAATAGAAACAGAAGCAAATTGGCGAGT</code></pre>
</div>
<div id="step-2-download-candidate-regions-of-enhancers-from-encode-by-h3k4me1-marks" class="section level2">
<h2 class="hasAnchor">
<a href="#step-2-download-candidate-regions-of-enhancers-from-encode-by-h3k4me1-marks" class="anchor"></a>Step 2, download candidate regions of enhancers from ENCODE by H3K4me1 marks</h2>
<p>By default, the hisone marker is H3K4me1. Users can also define the markers by <code>markers</code> parameter in the function <code>getENCODEdata</code>. To make sure the markers are tissue specific, we can filter the data by <code>biosample_name</code> and <code>biosample_type</code> parameters. For additional filters, please refer <code><a href="../reference/getENCODEdata.html">?getENCODEdata</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://jianhong.github.io/enhancerHomologSearch">enhancerHomologSearch</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span>
<span class="co"># download enhancer candidates for human heart tissue</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getENCODEdata.html">getENCODEdata</a></span><span class="op">(</span>genome<span class="op">=</span><span class="va">Hsapiens</span>,
                    partialMatch<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>biosample_summary <span class="op">=</span> <span class="st">"heart"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># download enhancer candidates for mouse heart tissue</span>
<span class="va">mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getENCODEdata.html">getENCODEdata</a></span><span class="op">(</span>genome<span class="op">=</span><span class="va">Mmusculus</span>,
                    partialMatch<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>biosample_summary <span class="op">=</span> <span class="st">"heart"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="step-3-get-alignment-score-for-target-enhancer-and-candidate-enhancers-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-3-get-alignment-score-for-target-enhancer-and-candidate-enhancers-" class="anchor"></a>Step 3, get alignment score for target enhancer and candidate enhancers.</h2>
<p>This step is time consuming step. For quick run, users can subset the data by given genomic coordinates.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset the data for test run </span>
<span class="co"># in human, the homolog HBEGF gene is located at chromosome 5</span>
<span class="co"># you can try to subset the data by chromosome 5</span>
<span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">Hsapiens</span><span class="op">)</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">hs</span>, <span class="va">gr</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">==</span><span class="st">"chr7"</span><span class="op">]</span><span class="op">)</span>
<span class="co"># In this test run, we will only use upstream 1M and downstream 1M of homolog</span>
<span class="co"># gene</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="va">eid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">mget</a></span><span class="op">(</span><span class="st">"LEP"</span>, <span class="va">org.Hs.egALIAS2EG</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">g_hs</span> <span class="op">&lt;-</span> <span class="fu">select</span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span>,
               keys<span class="op">=</span><span class="va">eid</span>,
               columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GENEID"</span>, <span class="st">"TXCHROM"</span>, <span class="st">"TXSTART"</span>, <span class="st">"TXEND"</span>, <span class="st">"TXSTRAND"</span><span class="op">)</span>,
               keytype<span class="op">=</span><span class="st">"GENEID"</span><span class="op">)</span>
<span class="va">g_hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">g_hs</span>, <span class="fu">GRanges</span><span class="op">(</span><span class="va">TXCHROM</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="va">TXSTART</span>, <span class="va">TXEND</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">expandGR</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">ext</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">-</span><span class="va">ext</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">+</span><span class="va">ext</span>
  <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">trim</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">hs</span>, <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_hs</span>, ext<span class="op">=</span><span class="fl">1000000</span><span class="op">)</span><span class="op">)</span>
<span class="co"># in mouse, the homolog Hbegf gene is located at chromosome 18</span>
<span class="co"># same as above script in human, you can try to subset the data by chromosome.</span>
<span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">Mmusculus</span><span class="op">)</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="va">mm</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">mm</span>, <span class="va">gr</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">==</span><span class="st">"chr6"</span><span class="op">]</span><span class="op">)</span>
<span class="co"># Here we use the subset of 1M upstream and downstream of homolog gene.</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Mmusculus.UCSC.mm10.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span>
<span class="va">eid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">mget</a></span><span class="op">(</span><span class="st">"Lep"</span>, <span class="va">org.Mm.egALIAS2EG</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">g_mm</span> <span class="op">&lt;-</span> <span class="fu">select</span><span class="op">(</span><span class="va">TxDb.Mmusculus.UCSC.mm10.knownGene</span>,
               keys<span class="op">=</span><span class="va">eid</span>,
               columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GENEID"</span>, <span class="st">"TXCHROM"</span>, <span class="st">"TXSTART"</span>, <span class="st">"TXEND"</span>, <span class="st">"TXSTRAND"</span><span class="op">)</span>,
               keytype<span class="op">=</span><span class="st">"GENEID"</span><span class="op">)</span>
<span class="va">g_mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">g_mm</span>,
                   <span class="fu">GRanges</span><span class="op">(</span><span class="va">TXCHROM</span>,
                           <span class="fu">IRanges</span><span class="op">(</span><span class="va">TXSTART</span>, <span class="va">TXEND</span><span class="op">)</span>,
                           strand<span class="op">=</span><span class="va">TXSTRAND</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">g_mm</span> <span class="op">&lt;-</span> <span class="va">g_mm</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">g_mm</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"chr6"</span> <span class="op">&amp;</span> <span class="fu">strand</span><span class="op">(</span><span class="va">g_mm</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"+"</span><span class="op">]</span>
<span class="va">mm</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">mm</span>, <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_mm</span>, ext<span class="op">=</span><span class="fl">1000000</span><span class="op">)</span><span class="op">)</span>

<span class="co"># use parallel computing to speed up.</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel">BiocParallel</a></span><span class="op">)</span>
<span class="va">bpparam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html">MulticoreParam</a></span><span class="op">(</span>tasks<span class="op">=</span><span class="fl">200</span>, progressbar<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">aln_hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignmentOne.html">alignmentOne</a></span><span class="op">(</span><span class="va">seqEN</span>, <span class="va">hs</span>, bpparam<span class="op">=</span><span class="va">bpparam</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aln_mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignmentOne.html">alignmentOne</a></span><span class="op">(</span><span class="va">seqEN</span>, <span class="va">mm</span>, bpparam<span class="op">=</span><span class="va">bpparam</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div id="step-4-filter-the-candidate-regions-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-4-filter-the-candidate-regions-" class="anchor"></a>Step 4, filter the candidate regions.</h2>
<p>Here we will filter the candidate regions more than 5K from TSS of homolog but within 100K from the gene body. The candidates will be also filtered by p-value.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Step4</span>
<span class="va">ext</span> <span class="op">&lt;-</span> <span class="fl">100000</span>
<span class="va">aln_hs</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">aln_hs</span>, ranges <span class="op">=</span> <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_hs</span>, ext<span class="op">=</span><span class="va">ext</span><span class="op">)</span><span class="op">)</span>
<span class="co">## filter by distance</span>
<span class="fu"><a href="../reference/Enhancers-class.html">distance</a></span><span class="op">(</span><span class="va">aln_hs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Enhancers-class.html">distance</a></span><span class="op">(</span><span class="fu"><a href="../reference/Enhancers-class.html">peaks</a></span><span class="op">(</span><span class="va">aln_hs</span><span class="op">)</span>, <span class="va">g_hs</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">aln_hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">aln_hs</span>, <span class="va">pval</span><span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">distance</span> <span class="op">&gt;</span><span class="fl">5000</span><span class="op">)</span>


<span class="va">aln_mm</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">aln_mm</span>, ranges <span class="op">=</span> <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_mm</span>, ext<span class="op">=</span><span class="va">ext</span><span class="op">)</span><span class="op">)</span>
<span class="co">## filter by distance</span>
<span class="fu"><a href="../reference/Enhancers-class.html">distance</a></span><span class="op">(</span><span class="va">aln_mm</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Enhancers-class.html">distance</a></span><span class="op">(</span><span class="fu"><a href="../reference/Enhancers-class.html">peaks</a></span><span class="op">(</span><span class="va">aln_mm</span><span class="op">)</span>, <span class="va">g_mm</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">aln_mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">aln_mm</span>, <span class="va">pval</span><span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">distance</span> <span class="op">&gt;</span><span class="fl">5000</span><span class="op">)</span></code></pre></div>
</div>
<div id="step-5-export-the-multiple-alignments-in-order-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-5-export-the-multiple-alignments-in-order-" class="anchor"></a>Step 5, export the multiple alignments in order.</h2>
<p>The selected candidates will be aligned cross human and mouse and then output as phylip multiple alignment file in text format.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">al</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignment.html">alignment</a></span><span class="op">(</span><span class="va">seqEN</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>human<span class="op">=</span><span class="va">aln_hs</span>, mouse<span class="op">=</span><span class="va">aln_mm</span><span class="op">)</span>,
                method<span class="op">=</span><span class="st">"ClustalW"</span>, order<span class="op">=</span><span class="st">"input"</span><span class="op">)</span>
<span class="va">al</span></code></pre></div>
<pre><code>## [[1]]
## DNAMultipleAlignment with 3 rows and 1686 columns
##      aln                                                    names               
## [1] TGGCATACACAGCAAACATCATGAAT...TAGAAACAGAAGCAAATTGGCGAGT Enhancer
## [2] --------------------------...------------------------- human_chr7:128340...
## [3] --------------------------...------------------------- mouse_chr6:291629...</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MotifDb</span><span class="op">)</span>
<span class="va">motifs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MotifDb/man/query.html">query</a></span><span class="op">(</span><span class="va">MotifDb</span>, <span class="st">"JASPAR_CORE"</span><span class="op">)</span>
<span class="va">consensus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">motifs</span>, <span class="va">consensusString</span><span class="op">)</span>
<span class="va">consensus</span> <span class="op">&lt;-</span> <span class="fu">DNAStringSet</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\?"</span>, <span class="st">"N"</span>, <span class="va">consensus</span><span class="op">)</span><span class="op">)</span>
<span class="va">tmpfolder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="../reference/saveAlignments.html">saveAlignments</a></span><span class="op">(</span><span class="va">al</span>, output_folder <span class="op">=</span> <span class="va">tmpfolder</span>, motifConsensus<span class="op">=</span><span class="va">consensus</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/readLines.html">readLines</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">tmpfolder</span>, <span class="st">"aln1.phylip.txt"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   [1] " 5 1686"                                                                                  
##   [2] "Enhancer                           TGGCATACAC AGCAAACATC ATGAATTTAA TTTAATTTAA TTTAATTTAA"
##   [3] "human_chr7:128340906-128341905:-   ---------- ---------- ---------- ---------- ----------"
##   [4] "mouse_chr6:29162916-29163915:+     ---------- ---------- ---------- ---------- ----------"
##   [5] "Consensus                          ---------- ---------- ---------- ---------- ----------"
##   [6] "motifConsensus                     ---------- ---------- ---------- ---------- ----------"
##   [7] ""                                                                                         
##   [8] "                                   TTTAATTTTT TTAATTTAAT TTTAATATTT TAAAATAAAA TAAAATAAAA"
##   [9] "                                   ---------- ---------- ---------- ---------- ----------"
##  [10] "                                   ---------- ---------- ---------- ---------- ----------"
##  [11] "                                   ---------- ---------- ---------- ---------- ----------"
##  [12] "                                   ---------- ---------- ---------- ---------- ----------"
##  [13] ""                                                                                         
##  [14] "                                   TAAAATAAAA TAAAAGATAA AGATAAAGAT AAAATAAAAT TCAACTCAAT"
##  [15] "                                   ---------- ---------- ---------- ---------- ----------"
##  [16] "                                   ---------- ---------- ---------- ---------- ----------"
##  [17] "                                   ---------- ---------- ---------- ---------- ----------"
##  [18] "                                   ---------- ---------- ---------- ---------- ----------"
##  [19] ""                                                                                         
##  [20] "                                   TAAATTAAAA CTAAGCTAAA ATAAAAATAC AATAAAATAA ATTTCAATTT"
##  [21] "                                   ---------- ---------- ---------- -------GGT GTTAGTAGCT"
##  [22] "                                   ---------- ---------- ---------- ---------- ----------"
##  [23] "                                   ---------- ---------- ---------- ---------- -TT---A--T"
##  [24] "                                   ---------- ---------- ---------- ---------- ----------"
##  [25] ""                                                                                         
##  [26] "                                   AATGTAATTT AATTTAAAAA GGGACTACGC CGAAAAGAAA ATGAATGAAT"
##  [27] "                                   GAATTAACTC CTCCTCACCA GCCCCCATGT TCTTCCATAG --GACTCCAC"
##  [28] "                                   ---------- ---------- ---------- ---------- ----------"
##  [29] "                                   -A--TAA-T- ----T-A--A G---C-A-G- --------A- --GA-T--A-"
##  [30] "                                   ---------- ---------- ---------- ---------- ----------"
##  [31] ""                                                                                         
##  [32] "                                   GGATGAATAA ATAATTTAAT TTAATTTAAT TTAATTTAAT TTAATTTAAT"
##  [33] "                                   AAA-GAATAA T--ATTAGGA CCAAGGAGAT GAAAACTACA AATATCAACT"
##  [34] "                                   ---------- ---ACAACGT CTTATATAAC CCAAGTAGGC CTCAACT-TC"
##  [35] "                                   --A-GAATAA ---ATT---T -TAAT-TAAT --AA-TTA-- -T-AT-TA-T"
##  [36] "                                   ---------- ---------- ---------- ---------- ----------"
##  [37] ""                                                                                         
##  [38] "                                   TTAATTTAAT TTAATTTAAT TTAATTTAAT TTAATTTAAT TTAATTTAAT"
##  [39] "                                   GTAACCCACC TTTGCCCTTT TTGGTTTAAC ---ATTTTGT ATAGTAAAAT"
##  [40] "                                   CTGATCCTCC TGCCTCCACT T--ACTAAGT ---GCTGGGA TTACAGTCAG"
##  [41] "                                   -TAAT--A-- TT--T--A-T TT-ATTTAAT ---ATTT--T TTA-T-TAAT"
##  [42] "                                   ---------- ---------- ---------- ---------- ----------"
##  [43] ""                                                                                         
##  [44] "                                   TTAATTTAAT TTGTTCGGCA CAGTAT-AAT ATGCTAGCAT CTCAGTTATT"
##  [45] "                                   TTTAAATAAT TTATT--GCA TCAAAA-AGG AAAAGGACAG ATCAAGAA--"
##  [46] "                                   ATACCACCAC TCCTG--GCA CAACATTAGT GTTGTAACAT -TTATTTA--"
##  [47] "                                   TTAA--TAAT TT-TT--GCA CA--AT-A-T AT--TA-CAT -TCA-TTA--"
##  [48] "                                   ---------- ---------- ---------- ---------- ----------"
##  [49] ""                                                                                         
##  [50] "                                   TCACGTGTGT TGTTACTATA AAATAAGCAA AACAGTGATA AAATAAGTTT"
##  [51] "                                   ----GTCTGC CTTTACCTGA AAAGGAC--- -------ACA TAGTTCATAT"
##  [52] "                                   ---CTTATGT GTGTGCAAAA GTATGGG--- -------AAA AATCTGGTAG"
##  [53] "                                   ---CGT-TGT --TTAC-A-A AAAT-AG--- -------A-A AA-T--GT-T"
##  [54] "                                   ---------- ---------- ---------- ---------- ----------"
##  [55] ""                                                                                         
##  [56] "                                   GTGTTGCTTA TCTTATGACT GG--TGGAAT GTAACAGGGA AAAAAAGCAC"
##  [57] "                                   GCTCTGCTTC TCTTTCCACT GCATTTGAAT CTCCTTCAGG GTATAAGCCC"
##  [58] "                                   GAGTTGGTTC T-TTCTGTCT AT--CAGGTT GAAGTGGGTC CTGAGATCTC"
##  [59] "                                   G-GTTGCTT- TCTT-TGACT G---T-GAAT GTA---GGG- --AAAAGC-C"
##  [60] "                                   ---------- ---------- ---------- ---------- ---AAAGCN-"
##  [61] ""                                                                                         
##  [62] "                                   ATACTGTGAC TTTGACAAAA CTGAGTGACT GATGATAATA AACTTCTCTT"
##  [63] "                                   ---------- TTTGACACAG ----GACACT GATCTACATA AACTGGT---"
##  [64] "                                   TT-------- CTCGATGTAC TC--GTGAAA GTTGGACATA ATTTATTCTT"
##  [65] "                                   -T-------- TTTGACA-A- ----GTGACT GATG---ATA AACT--TCTT"
##  [66] "                                   ---------N TTTGACA--- ---------- ---------- ----------"
##  [67] ""                                                                                         
##  [68] "                                   CTCGTAAG-C TGACAGTTCA TAAAACCTCT GCTTGTTTTT TTGTACTTTT"
##  [69] "                                   ----TAAA-G CAACATAACA TCAA---TTT TTCTCTTCTC TTGAACTCCT"
##  [70] "                                   CTGCCAAGTC TTTCATTTTA TCTG---TCT CTCAATTAGA TATTACTTTC"
##  [71] "                                   CT--TAAG-C T-ACA-TTCA T-AA---TCT ---T-TT-T- TTGTACTTTT"
##  [72] "                                   ---------- ---------- ---------- ---------- ----------"
##  [73] ""                                                                                         
##  [74] "                                   AATCTTAAGG TGACGCATGT AGCTTCCTGT CCTTCTCAGT TTACTGACAG"
##  [75] "                                   TATCATAA-- ------AAGC TGCCCACTCA CCTT-TCAGA AAATGAGCAC"
##  [76] "                                   AAACATGG-- ------GTGC TTCCTTCTAC AATT----GT TTGGGATTAA"
##  [77] "                                   AATC-TAA-- ------ATG- -GC-T-CT-- CCTT-TCAGT TTA----CA-"
##  [78] "                                   ---------- ---------- ---------- ---------- ----------"
##  [79] ""                                                                                         
##  [80] "                                   AGGTTAGGGT TTAATCCCAG ATATCCAGTC TGACTGTACA GTAGTTCAGG"
##  [81] "                                   AAGAAAATTC ATAGCCCTTC ATGATCAGTA CAACCA-AAA GTAAAGCAGC"
##  [82] "                                   AGGTGTGTGC TGAAGC---- -TGAGCCACA CAACTAGAAA TAGGATTTTC"
##  [83] "                                   AGGT-AG-G- TTAA-CC--- AT---CAGT- --ACT--A-A GTAG-TCAG-"
##  [84] "                                   ---------- ---------- ---------- ---------- ----------"
##  [85] ""                                                                                         
##  [86] "                                   AGACCGACGC AGATTTATAG CATCATTCGT CAAACCCTGA GGATAATCAT"
##  [87] "                                   TGA--AATGT AAA-----AG AGTTTGACAC CCAAATGAAA GGACATACAG"
##  [88] "                                   TAGTAAACAC CAC------- AATCTGAGGA CTCAGTGTGA TCG-AATCTC"
##  [89] "                                   -GA---ACGC A-A-----AG -ATC---CG- C-AA---TGA GGA-AATCA-"
##  [90] "                                   ---------- ---------- ---------- ---------- ----------"
##  [91] ""                                                                                         
##  [92] "                                   TTGTCACAGC TTCCTTTGGT -CATCATTAC TGTG-CAAAT AAACTGTTAG"
##  [93] "                                   TAAATACCAC ACATTTTACA -TGTCTGCAT AAAG-CATGC TCAC--ATAC"
##  [94] "                                   CTGTCACACC TTCATTCACT TCCTCGTCAC CCTGGCCACC AAGCTCTCTT"
##  [95] "                                   TTGTCACA-C TTC-TTT--T -C-TC-T-AC --TG-CAA-- AAACT-TTA-"
##  [96] "                                   ---------- ---------- ---------- ---------- ----------"
##  [97] ""                                                                                         
##  [98] "                                   AGCATGAGCC AGCAAAAACA GTGGGAAACG CAGCAATTTC CTGTATTTAA"
##  [99] "                                   ATTTAGAGTC CCTGACAACC CTGTGAAATT A------TTC CCATCTTAAA"
## [100] "                                   CTCACACAAG GACATACTTG CTTTAATATC CTCAAGGTTG CTGTTTCCTG"
## [101] "                                   A-CA-GAG-C --CAAAAAC- -TG-GAAA-- C---A--TTC CTGT-TT-AA"
## [102] "                                   ---------- ---------- ---------- ---------- ----------"
## [103] ""                                                                                         
## [104] "                                   TAGTCTGTGA GATATACTTT AATGAGATGA AATTGAAGAA AACTGAGTCA"
## [105] "                                   GAGGTGAAGT GACATGCACT GCT------- -ATTGAGGAA GACT---TCA"
## [106] "                                   AATTCAGACT GTTTTATCCC CCC------A GTTCTTAGTT CACT--CTCC"
## [107] "                                   -AGTC-G-G- GATATAC--T --T------A -ATTGAAGAA -ACT---TCA"
## [108] "                                   ---------- -ATATA---- ---------- ---------- ----------"
## [109] ""                                                                                         
## [110] "                                   TTAGAAAGGC ATTCACATAA ACTTTCCTG- GTGTATATTT CCTAACTCTC"
## [111] "                                   TTCCACATAC CCTCCCTCCC GTTTCACTGA GTGCAGCACT CTTTCCAGTA"
## [112] "                                   TCAAACAGGT CTTCCCT--A GCCACCCTAT CTACACACAC ACATTCCCTC"
## [113] "                                   TTA-A-AGGC -TTC-C---A -CTT-CCTG- GTG-A-A--T CCT--C-CTC"
## [114] "                                   ---------- ---------- ---------- ---------- ----------"
## [115] ""                                                                                         
## [116] "                                   TTCCAGTGTT TTCTACACCA GAAGAGTTCA TTAC-ATCAT TGAAGGACAA"
## [117] "                                   CATATCAGCT GTCTGGAT-A GGAAAGTGCA TAGT-TTTTT AAAAGGAAG-"
## [118] "                                   TTT-TGACTT TTCTCCTTAG CATAAATTTA TCACCATTAT CAACATATAG"
## [119] "                                   TT---G-GTT TTCT-CA--A GAA-AGTTCA T-AC-AT-AT --AAGGA-A-"
## [120] "                                   ---------- ---------- ---------- ---------- ----------"
## [121] ""                                                                                         
## [122] "                                   TGCTGAAAAA TAAGAACGCG TTTGGTTTTT CATAAACCAC ATGGTCTTGT"
## [123] "                                   ---TAAGAGA TTTTTAAGGA TTTAG----- -ACCAAGTAC ACAAGGTGTA"
## [124] "                                   T-TTGGCTAT TTGTTGGCTG TCTTT----- --CAAACCAT AAGTGTATGA"
## [125] "                                   T--TGA-AAA T----A-G-G TTT-G----- -A-AAACCAC A-G---TTG-"
## [126] "                                   ---------- ---------- ---------- ---------- ----------"
## [127] ""                                                                                         
## [128] "                                   GGGTCATGT- TGTTTTGTTT CTTTAGATTT GAGAGACGGG GAATGATGTG"
## [129] "                                   AGGTAAGAT- TGCATGTAGC CTATATTCTG AGGACTCAAA GGAGAAC---"
## [130] "                                   GAACAGAATC TGCGTCAGTG ATTTAGTGTC -ACACACCTG TTGTATC---"
## [131] "                                   GGGT-A--T- TG--T---T- CTTTAG--T- -AGA-AC--G G-AT-A----"
## [132] "                                   ---------- ---------- ---------- ---------- ----------"
## [133] ""                                                                                         
## [134] "                                   ATTTTGCCCA GTCAGCATGG ATATGATTTG GACTTC-CAT CTGTTTAAGA"
## [135] "                                   -TCCCATTCA AT-GGCCTAT -TCTTCCCTG GAATAC-AAC CACCTGGGTA"
## [136] "                                   -CTAAGGGCA GCCAGCAATG TTCTAAAATA CAGTTGGCAT CCAACAAGTA"
## [137] "                                   -TT--G--CA GTCAGCAT-G -T-T-A--TG GA-TTC-CAT C---T-A--A"
## [138] "                                   ---------- ---------- ---------- ---------- ----------"
## [139] ""                                                                                         
## [140] "                                   TTAAATGGTA GACAGAGAGA AATATTTCTG TTTTTTTTAT CCATGATTGC"
## [141] "                                   CGGAATGCTA CTTACACTGG AACCCAGTAC ATACATATAT TCTTTATTTA"
## [142] "                                   ATAAATAA-A TAAACAGATG AATGCATAGG ATGTGCTAAC GTGCTAATCA"
## [143] "                                   -TAAATG-TA -A-A-AGAG- AAT---T--G -T-T-TTTAT -C-T-ATT--"
## [144] "                                   ---------- ---------- ---------- ---------- ----------"
## [145] ""                                                                                         
## [146] "                                   AAATCTGTGG GTT---CAA- -AGTCTGCTT TTGTTCCAAA TAATCATTC-"
## [147] "                                   AAACTTAAAA GTTTTACAA- -ATACTTATC CTTTACCATA TATGGACGT-"
## [148] "                                   AACTGAGTCC TCTCACTAGC CAGATCAACT CCAGACCATT TAATGTCTTT"
## [149] "                                   AAAT-TGT-- GTT---CAA- -AG-CT--TT -T-T-CCA-A TAAT-A-T--"
## [150] "                                   ---------- ---------- ---------- ---------- ----------"
## [151] ""                                                                                         
## [152] "                                   -AAACCTGCC GTACTGTGTG GGGTGGGAAG TGAAGGAGGA TCTTATCTGG"
## [153] "                                   -ACTCCTATA CTATTTAATT TTTTAAGAAC TCTAGGCCAG GCGTGGTGGC"
## [154] "                                   AAAGATTATA TTCCTTTATT TAA-AAAAAA TGGGGTTAAG GCTCTTTGTC"
## [155] "                                   -AA-CCT--- -TACT-T-T- ---T--GAA- TG-AGG---- -CTT-T--G-"
## [156] "                                   ---------- ---------- ---------- ---------- ----------"
## [157] ""                                                                                         
## [158] "                                   AAATCATGTG CTGTATGATG AAGGCAGGAT ATGGAAAACT CCAAATATGG"
## [159] "                                   A--------- ---------- ---------- ---------- ----------"
## [160] "                                   CTATAAGATT GTTGAGAACA TTAAGTAAGT TAACCCATGT CAAACTAAGC"
## [161] "                                   A-AT-A--T- -T--A--A-- ---------T ------A--T C-AA-TA-G-"
## [162] "                                   ---------- ---------- ---------- ---------- ----------"
## [163] ""                                                                                         
## [164] "                                   ACACCTTTAT GTGTGCAAGG GAGAAAGTCT GAAGGATGCA ACCTGTTCAT"
## [165] "                                   ---------- ---------- ---------- ---------- ----------"
## [166] "                                   ACTC------ ---------- ---------- ---------- ----------"
## [167] "                                   AC-C------ ---------- ---------- ---------- ----------"
## [168] "                                   ---------- ---------- ---------- ---------- ----------"
## [169] ""                                                                                         
## [170] "                                   AACATTTTCA TTCAAATTTA AACTAGTTTG ATTAATTCCA AATGCACATT"
## [171] "                                   ---------- ---------- ---------- ---------- ----------"
## [172] "                                   ---------- ---------- ---------- ---------- ----------"
## [173] "                                   ---------- ---------- ---------- ---------- ----------"
## [174] "                                   ---------- ---------- ---------- ---------- ----------"
## [175] ""                                                                                         
## [176] "                                   TGATTTGTTG TGTTTTTATG ATGTATTTCA CAATACTGTT GCATAAAATA"
## [177] "                                   ---------- ---------- ---------- ---------- ----------"
## [178] "                                   ---------- ---------- ---------- ---------- ----------"
## [179] "                                   ---------- ---------- ---------- ---------- ----------"
## [180] "                                   ---------- ---------- ---------- ---------- ----------"
## [181] ""                                                                                         
## [182] "                                   TCTAAAAAAA ACATTTAGTT ATATGGAAGA CACTTGGACA ACTGGTTGTT"
## [183] "                                   ---------- ---------- ---------- ---------- ----------"
## [184] "                                   ---------- ---------- ---------- ---------- ----------"
## [185] "                                   ---------- ---------- ---------- ---------- ----------"
## [186] "                                   ---------- ---------- ---------- ---------- ----------"
## [187] ""                                                                                         
## [188] "                                   ATTTGTTTGT CTATTTTTAT GAATGCCTCA AAGATCAAAT AGTTACACAC"
## [189] "                                   ---------- ---------- ---------- ---------- ----------"
## [190] "                                   ---------- ---------- ---------- ---------- ----------"
## [191] "                                   ---------- ---------- ---------- ---------- ----------"
## [192] "                                   ---------- ---------- ---------- ---------- ----------"
## [193] ""                                                                                         
## [194] "                                   TTAATGCAAT CGAGCTTAGA GAGAGAAATT AAAAGTCTTA AATAAATTGT"
## [195] "                                   ---------- ---------- ---------- ---------- ----------"
## [196] "                                   ---------- ---------- ---------- ---------- ----------"
## [197] "                                   ---------- ---------- ---------- ---------- ----------"
## [198] "                                   ---------- ---------- ---------- ---------- ----------"
## [199] ""                                                                                         
## [200] "                                   GATTAGATAA ATAGAAACAG AAGCAAATTG GCGAGT"               
## [201] "                                   ---------- ---------- ---------- ------"               
## [202] "                                   ---------- ---------- ---------- ------"               
## [203] "                                   ---------- ---------- ---------- ------"               
## [204] "                                   ---------- ---------- ---------- ------"</code></pre>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.1 (2021-08-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.3 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] MotifDb_1.35.5                           
##  [2] org.Mm.eg.db_3.13.0                      
##  [3] TxDb.Mmusculus.UCSC.mm10.knownGene_3.10.0
##  [4] org.Hs.eg.db_3.13.0                      
##  [5] TxDb.Hsapiens.UCSC.hg38.knownGene_3.13.0 
##  [6] GenomicFeatures_1.45.2                   
##  [7] AnnotationDbi_1.55.1                     
##  [8] Biobase_2.53.0                           
##  [9] BSgenome.Mmusculus.UCSC.mm10_1.4.3       
## [10] BSgenome.Hsapiens.UCSC.hg38_1.4.3        
## [11] BSgenome.Drerio.UCSC.danRer10_1.4.2      
## [12] BSgenome_1.61.0                          
## [13] rtracklayer_1.53.1                       
## [14] Biostrings_2.61.2                        
## [15] XVector_0.33.0                           
## [16] GenomicRanges_1.45.0                     
## [17] GenomeInfoDb_1.29.8                      
## [18] IRanges_2.27.2                           
## [19] S4Vectors_0.31.3                         
## [20] BiocGenerics_0.39.2                      
## [21] BiocParallel_1.27.7                      
## [22] enhancerHomologSearch_0.99.12            
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7                matrixStats_0.60.1         
##  [3] fs_1.5.0                    bit64_4.0.5                
##  [5] filelock_1.0.2              progress_1.2.2             
##  [7] httr_1.4.2                  rprojroot_2.0.2            
##  [9] tools_4.1.1                 utf8_1.2.2                 
## [11] R6_2.5.1                    DBI_1.1.1                  
## [13] splitstackshape_1.4.8       withr_2.4.2                
## [15] tidyselect_1.1.1            prettyunits_1.1.1          
## [17] bit_4.0.4                   curl_4.3.2                 
## [19] compiler_4.1.1              textshaping_0.3.5          
## [21] xml2_1.3.2                  desc_1.3.0                 
## [23] DelayedArray_0.19.2         rappdirs_0.3.3             
## [25] pkgdown_1.6.1               systemfonts_1.0.2          
## [27] stringr_1.4.0               digest_0.6.27              
## [29] Rsamtools_2.9.1             rmarkdown_2.10             
## [31] pkgconfig_2.0.3             htmltools_0.5.2            
## [33] MatrixGenerics_1.5.4        dbplyr_2.1.1               
## [35] fastmap_1.1.0               rlang_0.4.11               
## [37] RSQLite_2.2.8               BiocIO_1.3.0               
## [39] generics_0.1.0              jsonlite_1.7.2             
## [41] dplyr_1.0.7                 RCurl_1.98-1.4             
## [43] magrittr_2.0.1              GenomeInfoDbData_1.2.6     
## [45] Matrix_1.3-4                Rcpp_1.0.7                 
## [47] fansi_0.5.0                 lifecycle_1.0.0            
## [49] stringi_1.7.4               yaml_2.2.1                 
## [51] SummarizedExperiment_1.23.4 zlibbioc_1.39.0            
## [53] BiocFileCache_2.1.1         grid_4.1.1                 
## [55] blob_1.2.2                  parallel_4.1.1             
## [57] crayon_1.4.1                lattice_0.20-44            
## [59] hms_1.1.0                   KEGGREST_1.33.0            
## [61] knitr_1.34                  pillar_1.6.2               
## [63] rjson_0.2.20                biomaRt_2.49.4             
## [65] XML_3.99-0.7                glue_1.4.2                 
## [67] evaluate_0.14               data.table_1.14.0          
## [69] vctrs_0.3.8                 png_0.1-7                  
## [71] purrr_0.3.4                 assertthat_0.2.1           
## [73] cachem_1.0.6                xfun_0.25                  
## [75] restfulr_0.0.13             ragg_1.1.3                 
## [77] tibble_3.1.4                GenomicAlignments_1.29.0   
## [79] memoise_2.0.0               ellipsis_0.3.2</code></pre>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-howe2013zebrafish" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Howe, K. <em>et al.</em> The zebrafish reference genome sequence and its relationship to the human genome. <em>Nature</em> <strong>496,</strong> 498–503 (2013).</div>
</div>
<div id="ref-braasch2016spotted" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Braasch, I. <em>et al.</em> The spotted gar genome illuminates vertebrate evolution and facilitates human-teleost comparisons. <em>Nature genetics</em> <strong>48,</strong> 427–437 (2016).</div>
</div>
<div id="ref-wong2020deep" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Wong, E. S. <em>et al.</em> Deep conservation of the enhancer regulatory code in animals. <em>Science</em> <strong>370,</strong> (2020).</div>
</div>
<div id="ref-pages2013package" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Pages, H., Aboyoun, P., Gentleman, R. &amp; DebRoy, S. Package <span>“biostrings”</span>. <em>R package version</em>
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jianhong Ou, Kenneth Poss.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
