<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>enhancerHomologSearch Guide • enhancerHomologSearch</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="enhancerHomologSearch Guide">
<meta property="og:description" content="enhancerHomologSearch">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">enhancerHomologSearch</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/enhancerHomologSearch.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="enhancerHomologSearch_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>enhancerHomologSearch Guide</h1>
                        <h4 class="author">Jianhong Ou</h4>
            
      
      
      <div class="hidden name"><code>enhancerHomologSearch.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>There is an increasing requirement for the tools to identify of putative mammalian orthologs to enhancers in species other than human and mouse, such as zebrafish, which is lacking whole genome comparison analysis data. Take zebrafish as an example, there are two major methods to identify the orthologs to enhancers in human and mouse,</p>
<ol style="list-style-type: decimal">
<li><p>use the whole genome comparison analysis data and conservation data<span class="citation"><sup>1</sup></span>,</p></li>
<li><p>use spotted gar genome as bridge genome to search the orthologs<span class="citation"><sup>2</sup></span>.</p></li>
</ol>
<p>Both of the methods will work well in the coding region. However, there is lacking of comparative data in distal regulation region such as enhancers and silencers.</p>
<p>In 2020, Emily S. Wong et. al. provide a new methods for identification of putative human orthologs to enhancers of zebrafish<span class="citation"><sup>3</sup></span>. They used the method to interrogate conserved syntenic regions and human and mouse using candidate sponge enhancer sequences. First, they looked for overlap with available functional genomics information. For example, they used mouse ENCODE data to infer enhancer activity based on histone marks in specific tissues. Second, they select the best-aligned region by whole genome alignment from the candidates regions for human and mouse as orthologs. This methods provide the possibility to search orthologs for enhancers or silencers even there is not genome comparative data available.</p>
<p>This package is modified from Wong’s methods and provide the easy to used script for researchers to quick search putative mammalian orthologs to enhancers. The modified algorithm is: The candidate regions were determined by ENCODE histone marks (default is H3K4me1) in specific tissue for humam and mouse<span class="citation"><sup>4</sup></span>. The mapping score were calculated by pairwise alignment between enhancer sequences and candidates by global sequence alignment<span class="citation"><sup>4</sup></span>. The Z-score were calculated from mapping score and then converted to P-value based on two-side test from a normal distribution. The candidates were filtered by p-value and distance from the TSS of target homologs. And then the top candidates from human and mouse were aligned to each other and exported as multiple alignments with given enhancer<span class="citation"><sup>5</sup></span>.</p>
</div>
<div id="step-1-prepare-target-enhancer-sequences-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-1-prepare-target-enhancer-sequences-" class="anchor"></a>Step 1, prepare target enhancer sequences.</h2>
<p>In this example, we will use an enhancer of hbegf gene in zebrafish.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load genome sequences</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Drerio.UCSC.danRer10</span><span class="op">)</span>
<span class="co"># define the enhancer genomic coordinates</span>
<span class="va">hbegfEN</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"chr14"</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="fl">6760805</span>,    <span class="fl">6761115</span><span class="op">)</span><span class="op">)</span>
<span class="co"># extract the sequences as Biostrings::DNAStringSet object</span>
<span class="op">(</span><span class="va">seqEN</span> <span class="op">&lt;-</span> <span class="fu">getSeq</span><span class="op">(</span><span class="va">BSgenome.Drerio.UCSC.danRer10</span>, <span class="va">hbegfEN</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 1:
##     width seq
## [1]   311 GAGGGTTGATTACTCAGCCATTTGCCCCGAATGT...GAATTACACATCTGAATCAGAAGCAATTAAGCT</code></pre>
</div>
<div id="step-2-download-candidate-regions-of-enhancers-from-encode-by-h3k4me1-marks" class="section level2">
<h2 class="hasAnchor">
<a href="#step-2-download-candidate-regions-of-enhancers-from-encode-by-h3k4me1-marks" class="anchor"></a>Step 2, download candidate regions of enhancers from ENCODE by H3K4me1 marks</h2>
<p>By default, the hisone marker is H3K4me1. Users can also define the markers by <code>markers</code> parameter in the function <code>getENCODEdata</code>. To make sure the markers are tissue specific, we can filter the data by <code>biosample_name</code> and <code>biosample_type</code> parameters. For additional filters, please refer <code><a href="../reference/getENCODEdata.html">?getENCODEdata</a></code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># load library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">enhancerHomologSearch</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg38</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Mmusculus.UCSC.mm10</span><span class="op">)</span>
<span class="co"># download enhancer candidates for human heart tissue</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getENCODEdata.html">getENCODEdata</a></span><span class="op">(</span>genome<span class="op">=</span><span class="va">Hsapiens</span>,
                    partialMatch<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>biosample_summary <span class="op">=</span> <span class="st">"heart"</span><span class="op">)</span><span class="op">)</span>
<span class="co"># download enhancer candidates for mouse heart tissue</span>
<span class="va">mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getENCODEdata.html">getENCODEdata</a></span><span class="op">(</span>genome<span class="op">=</span><span class="va">Mmusculus</span>,
                    partialMatch<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>biosample_summary <span class="op">=</span> <span class="st">"heart"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="step-3-get-alignment-score-for-target-enhancer-and-candidate-enhancers-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-3-get-alignment-score-for-target-enhancer-and-candidate-enhancers-" class="anchor"></a>Step 3, get alignment score for target enhancer and candidate enhancers.</h2>
<p>This step is time consuming step. For quick run, users can subset the data by given genomic coordinates.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># subset the data for test run </span>
<span class="co"># in human, the homolog HBEGF gene is located at chromosome 5</span>
<span class="co"># you can try to subset the data by chromosome 5</span>
<span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">Hsapiens</span><span class="op">)</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">hs</span>, <span class="va">gr</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">==</span><span class="st">"chr5"</span><span class="op">]</span><span class="op">)</span>
<span class="co"># In this test run, we will only use upstream 1M and downstream 1M of homolog</span>
<span class="co"># gene</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="va">eid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">mget</a></span><span class="op">(</span><span class="st">"HBEGF"</span>, <span class="va">org.Hs.egALIAS2EG</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">g_hs</span> <span class="op">&lt;-</span> <span class="fu">select</span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg38.knownGene</span>,
               keys<span class="op">=</span><span class="va">eid</span>,
               columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GENEID"</span>, <span class="st">"TXCHROM"</span>, <span class="st">"TXSTART"</span>, <span class="st">"TXEND"</span>, <span class="st">"TXSTRAND"</span><span class="op">)</span>,
               keytype<span class="op">=</span><span class="st">"GENEID"</span><span class="op">)</span>
<span class="va">g_hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">g_hs</span>, <span class="fu">GRanges</span><span class="op">(</span><span class="va">TXCHROM</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="va">TXSTART</span>, <span class="va">TXEND</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">expandGR</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">ext</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/stats/start.html">start</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">-</span><span class="va">ext</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/start.html">end</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">+</span><span class="va">ext</span>
  <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">trim</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="op">}</span>
<span class="va">hs</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">hs</span>, <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_hs</span>, ext<span class="op">=</span><span class="fl">1000000</span><span class="op">)</span><span class="op">)</span>
<span class="co"># in mouse, the homolog Hbegf gene is located at chromosome 18</span>
<span class="co"># same as above script in human, you can try to subset the data by chromosome.</span>
<span class="va">gr</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">Mmusculus</span><span class="op">)</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="va">mm</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">mm</span>, <span class="va">gr</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span><span class="op">==</span><span class="st">"chr18"</span><span class="op">]</span><span class="op">)</span>
<span class="co"># Here we use the subset of 1M upstream and downstream of homolog gene.</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Mmusculus.UCSC.mm10.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Mm.eg.db</span><span class="op">)</span>
<span class="va">eid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html">mget</a></span><span class="op">(</span><span class="st">"Hbegf"</span>, <span class="va">org.Mm.egALIAS2EG</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">g_mm</span> <span class="op">&lt;-</span> <span class="fu">select</span><span class="op">(</span><span class="va">TxDb.Mmusculus.UCSC.mm10.knownGene</span>,
               keys<span class="op">=</span><span class="va">eid</span>,
               columns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GENEID"</span>, <span class="st">"TXCHROM"</span>, <span class="st">"TXSTART"</span>, <span class="st">"TXEND"</span>, <span class="st">"TXSTRAND"</span><span class="op">)</span>,
               keytype<span class="op">=</span><span class="st">"GENEID"</span><span class="op">)</span>
<span class="va">g_mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">g_mm</span>, <span class="fu">GRanges</span><span class="op">(</span><span class="va">TXCHROM</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="va">TXSTART</span>, <span class="va">TXEND</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">mm</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">mm</span>, <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_mm</span>, ext<span class="op">=</span><span class="fl">1000000</span><span class="op">)</span><span class="op">)</span>

<span class="co"># use parallel computing to speed up.</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/Bioconductor/BiocParallel">BiocParallel</a></span><span class="op">)</span>
<span class="va">bpparam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/MulticoreParam-class.html">MulticoreParam</a></span><span class="op">(</span>tasks<span class="op">=</span><span class="fl">200</span>, progressbar<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">aln_hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignmentOne.html">alignmentOne</a></span><span class="op">(</span><span class="va">seqEN</span>, <span class="va">hs</span>, bpparam<span class="op">=</span><span class="va">bpparam</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |======================================================================| 100%</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">aln_mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignmentOne.html">alignmentOne</a></span><span class="op">(</span><span class="va">seqEN</span>, <span class="va">mm</span>, bpparam<span class="op">=</span><span class="va">bpparam</span><span class="op">)</span></code></pre></div>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div id="step-4-filter-the-candidate-regions-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-4-filter-the-candidate-regions-" class="anchor"></a>Step 4, filter the candidate regions.</h2>
<p>Here we will filter the candidate regions more than 5K from TSS of homolog but within 100K from the gene body. The candidates will be also filtered by p-value.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Step4</span>
<span class="va">ext</span> <span class="op">&lt;-</span> <span class="fl">100000</span>
<span class="va">aln_hs</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">aln_hs</span>, ranges <span class="op">=</span> <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_hs</span>, ext<span class="op">=</span><span class="va">ext</span><span class="op">)</span><span class="op">)</span>
<span class="co">## filter by distance</span>
<span class="va">aln_hs</span><span class="op">$</span><span class="va">peaks</span><span class="op">$</span><span class="va">distance</span> <span class="op">&lt;-</span> <span class="fu">distance</span><span class="op">(</span><span class="va">aln_hs</span><span class="op">$</span><span class="va">peaks</span>, <span class="va">g_hs</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">aln_hs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">aln_hs</span>, <span class="va">pval</span><span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">distance</span> <span class="op">&gt;</span><span class="fl">5000</span><span class="op">)</span>


<span class="va">aln_mm</span> <span class="op">&lt;-</span> <span class="fu">subsetByOverlaps</span><span class="op">(</span><span class="va">aln_mm</span>, ranges <span class="op">=</span> <span class="fu">expandGR</span><span class="op">(</span><span class="va">g_mm</span>, ext<span class="op">=</span><span class="va">ext</span><span class="op">)</span><span class="op">)</span>
<span class="co">## filter by distance</span>
<span class="va">aln_mm</span><span class="op">$</span><span class="va">peaks</span><span class="op">$</span><span class="va">distance</span> <span class="op">&lt;-</span> <span class="fu">distance</span><span class="op">(</span><span class="va">aln_mm</span><span class="op">$</span><span class="va">peaks</span>, <span class="va">g_mm</span>, ignore.strand<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">aln_mm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">aln_mm</span>, <span class="va">pval</span><span class="op">&lt;</span><span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">distance</span> <span class="op">&gt;</span><span class="fl">5000</span><span class="op">)</span></code></pre></div>
</div>
<div id="step-5-export-the-multiple-alignments-in-order-" class="section level2">
<h2 class="hasAnchor">
<a href="#step-5-export-the-multiple-alignments-in-order-" class="anchor"></a>Step 5, export the multiple alignments in order.</h2>
<p>The selected candidates will be aligned cross human and mouse and then output as phylip multiple alignment file in text format.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">al</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/alignment.html">alignment</a></span><span class="op">(</span><span class="va">seqEN</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>human<span class="op">=</span><span class="va">aln_hs</span>, mouse<span class="op">=</span><span class="va">aln_mm</span><span class="op">)</span>,
                method<span class="op">=</span><span class="st">"ClustalOmega"</span>, order<span class="op">=</span><span class="st">"input"</span><span class="op">)</span></code></pre></div>
<pre><code>## using Gonnet
## using Gonnet
## using Gonnet
## using Gonnet
## using Gonnet
## using Gonnet
## using Gonnet
## using Gonnet
## using Gonnet
## using Gonnet</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tmpfolder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu"><a href="../reference/saveAlignments.html">saveAlignments</a></span><span class="op">(</span><span class="va">al</span>, output_folder <span class="op">=</span> <span class="va">tmpfolder</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/readLines.html">readLines</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">tmpfolder</span>, <span class="st">"aln1.phylip.txt"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   [1] " 4 1109"                                                                                  
##   [2] "Enhancer                           ---------- ---------- ---------- ---------- ----------"
##   [3] "human_chr5:140245727-140246726:+   CTCCTGGGAG TTCATCACAC ATCCCGAGAG GGAAGAGTGT CCTGGGCAGA"
##   [4] "mouse_chr18:36422863-36423862:-    GACATGGGAG CTCT------ ---------- ---------G GAAGAACAGA"
##   [5] "Consensus                          ---------- ---------- ---------- ---------- ----------"
##   [6] ""                                                                                         
##   [7] "                                   ---------- ---------- ---------- ---------- ----------"
##   [8] "                                   GGTGGCAGGC AAAGCCGGGT AAAAACTCCA GGGCTGGGAA GCAAATGGGG"
##   [9] "                                   AAGCACTCTC AGTGCTGGGT TAGCTCTCCA GCCCTCGTAG GAACTAC--T"
##  [10] "                                   ---------- ---------- ---------- ---------- ----------"
##  [11] ""                                                                                         
##  [12] "                                   ---------- ---------- ---------- ---------- ----------"
##  [13] "                                   CTCAGGGTGA TGCAGAAAAT GTGATGTTGC CAGGCCATCC AAATAAAGCA"
##  [14] "                                   TTCAAGCATA TGAACAT--- --------CT CTTTGACTTA AGTGACTGCT"
##  [15] "                                   ---------- ---------- ---------- ---------- ----------"
##  [16] ""                                                                                         
##  [17] "                                   ---------- ---------- ---------- ---------- ----------"
##  [18] "                                   TCCATCGGGG CAGAGGAGAA GCTGTTTCCC TGCAGACACT CCT-------"
##  [19] "                                   ACAAAAATAT GAAAGGAGAT GCTGTCATGC CATGAACACG CTTATTAATA"
##  [20] "                                   ---------- ---------- ---------- ---------- ----------"
##  [21] ""                                                                                         
##  [22] "                                   ---------- ---------- ---------- ---------- ----------"
##  [23] "                                   -----CTGCC CCCACCAGGA ATGGGAGGGG CAGGAGGAAG AGCTTCCCAG"
##  [24] "                                   CAGGTGTGCC CTTCCCCACA GTAAGTATTG AACACTCTGG CTTTTCTGGC"
##  [25] "                                   ---------- ---------- ---------- ---------- ----------"
##  [26] ""                                                                                         
##  [27] "                                   GAGGGTTGAT TACTCAGCCA TTTGCCCCGA ATGTCTAAGC GTGAGTCAGA"
##  [28] "                                   AGAGGCTCCC TACTGGGCCC TTCGTGCCAT CAG------- --------CA"
##  [29] "                                   CAGTACCTTT TTGTCTGTCT GTCCATCTGT CTGTCTAAAA CAGGATCTCA"
##  [30] "                                   -AGGG-T--T TACTC-GCC- TT-G--CCG- -TGTCTAA-- --G--TC--A"
##  [31] ""                                                                                         
##  [32] "                                   GCACCCAGCC TTTGTG-TCT CCATCTGAAA CCTTTTCCTC CCCTGTT---"
##  [33] "                                   TCTCCCGGAT GTTGTCCTCA GCTTCCTTAA CGCTTCGCTC CAGGTAG---"
##  [34] "                                   CATAGCCCTG ACTGTCCTGT AACTCACATA GCAAAGGCAA ACCTTCAACT"
##  [35] "                                   -C-CCC-G-- -TTGT--TCT -C-TC--AAA CC-TT--CTC CCCT------"
##  [36] ""                                                                                         
##  [37] "                                   -GTGTTTTTT TGGGGCTG-- ---------- ---------A TGAGCAGAT-"
##  [38] "                                   -GACTTTTTC TGCTGCAATA TGAGAGACAG ACAAAGGTTA GGACCAGAGT"
##  [39] "                                   TCTGATCCCC TGCTCTTGCC TCCCACAGTG GTAGGGCTAT CGATGTGC--"
##  [40] "                                   -GTGTTTTT- TG--GCTG-- ---------- ---------A -GA-CAGA--"
##  [41] ""                                                                                         
##  [42] "                                   GAATGGGCCC T----GGGGA CAAAGAGATG ATGTAGGTTT CTACGCATCT"
##  [43] "                                   GAATGGGCCG GGCTGGAAGA CACAGCTTTG ATGTC----- CAATGGTTAT"
##  [44] "                                   ACCAAGCCCC GGCTTGGCCA G-CATCTCTC ACTTGAAGCT GGATATCTCC"
##  [45] "                                   GAATGGGCCC -----GG-GA CA-AG---TG ATGT-----T C-A-G--TCT"
##  [46] ""                                                                                         
##  [47] "                                   GGCCCTTCAT CAATGGACTC GGATCAAACA CTCACCACAA TATATGCAAC"
##  [48] "                                   GGCTTTTCCT TTGTGACA-- ---GGCAACT TTCAGGAGTA TACTGCAAGC"
##  [49] "                                   TGCCCTGCAA GAACAGCAAC TGAGAGAATG AGAAC----- -TGGGTAAGT"
##  [50] "                                   GGCCCTTCAT -AATGG---C -GA---AAC- -TCAC-A--A TA-----A-C"
##  [51] ""                                                                                         
##  [52] "                                   CA---AGGCA ---------- -AATGTT--- ---------- -----CAGCT"
##  [53] "                                   CAGGTACACA CACCTGCCCA CAGTGCTCCC CTCACCTCCG AACCATGGGT"
##  [54] "                                   GAAGTCCACG GA-------- ----CTCGCG ATTACCACAG ----GGAGCT"
##  [55] "                                   CA---A--CA ---------- -A-TGTT--- ---------- ------AGCT"
##  [56] ""                                                                                         
##  [57] "                                   GATGATCTGC ATTCTGCTG- TGAGCCAGTT GATGAAATC- GCTGAA----"
##  [58] "                                   GATCACGGGC AGGCAGCAG- GGAGAAGATG GCTTTGGTA- GAAGCAAAAG"
##  [59] "                                   GCTCACTAAC ATGAAAGAAA ACTGAAAGTG TAACTTATCA CAAGAAACAA"
##  [60] "                                   GAT-A---GC AT-C-GC-G- -GAG--AGT- GAT---ATC- G--GAA----"
##  [61] ""                                                                                         
##  [62] "                                   ---------- ---------- -TTACACATC TGAATCAGAA GCAATTAAGC"
##  [63] "                                   TGCTCTGTGG AGGACAGGCA GTAGCACTCT TCCCCCAGAA GGAAGAAGAC"
##  [64] "                                   GCCTCTAGGC AGCCTGAG-A GCTATCTGAA GACCTCTCAA GC-ACAGGGC"
##  [65] "                                   ---------- ---------- -TTACAC--- T---TCAGAA GCAA--A-GC"
##  [66] ""                                                                                         
##  [67] "                                   T--------- ---------- ---------- ---------- ----------"
##  [68] "                                   TCTAAAAAAC AGGCGTCTCA GTACATGAAG TTATCCCAGA CAGACAAAGG"
##  [69] "                                   TGAGGAAGGT CGGGCTCCTG TACTCGTAAG TGGCCACAGT GTG-------"
##  [70] "                                   T--------- ---------- ---------- ---------- ----------"
##  [71] ""                                                                                         
##  [72] "                                   ---------- ---------- ---------- ---------- ----------"
##  [73] "                                   CCAGGAAATT CCAGGACAAA GCTCCTGTCC GTAACAGACT ATC----TCC"
##  [74] "                                   ---------T AACAGCCCAG GCTGCAGTCA GTAACTCCCT GAAGTGCTCA"
##  [75] "                                   ---------- ---------- ---------- ---------- ----------"
##  [76] ""                                                                                         
##  [77] "                                   ---------- ---------- ---------- ---------- ----------"
##  [78] "                                   TCTCAAGGTT TTCCATGGCC TGGGCCAGGC GTCTGTAGCC ACTGCAGCTA"
##  [79] "                                   AAAGGAGATC TGATAGGGGC TCAAACACTT CGAGGACGA- -------CAC"
##  [80] "                                   ---------- ---------- ---------- ---------- ----------"
##  [81] ""                                                                                         
##  [82] "                                   ---------- ---------- ---------- ---------- ----------"
##  [83] "                                   TAAAATGGAT GGTGCACGCT CTGCCCTGTC CAGCAGGAAT ATGGGCTGCT"
##  [84] "                                   AAAGCTCCCT AGAATTAGAT CTGAGAGGAA AAAACACAAC ATGCATTAAG"
##  [85] "                                   ---------- ---------- ---------- ---------- ----------"
##  [86] ""                                                                                         
##  [87] "                                   ---------- ---------- ---------- ---------- ----------"
##  [88] "                                   GTGAG----- -GGCTGTGGG GCTGTCTCGC ATCGCACTG- -----AGTGG"
##  [89] "                                   ATGCAGGAAG CTCATCTCGG GGTTCATTGC AGTGCACATG CAATTACTGC"
##  [90] "                                   ---------- ---------- ---------- ---------- ----------"
##  [91] ""                                                                                         
##  [92] "                                   ---------- ---------- ---------- ---------- ----------"
##  [93] "                                   ---------- ---------- ---------G TGAG---TCA GAAGGCAATC"
##  [94] "                                   AAATTTTTTA CCTTAACTGT AAAATAAAGG TACACCATCA GGGAACAGAG"
##  [95] "                                   ---------- ---------- ---------- ---------- ----------"
##  [96] ""                                                                                         
##  [97] "                                   ---------- ---------- ---------- ---------- ----------"
##  [98] "                                   -----TGCTT TTCTAATGAG ACTGCTGCCT GCCAGGAATC TGGGACACTA"
##  [99] "                                   TGAATCACAG GTGTTGTGTG TAAAATACCT ATGAAGTGTA TCTGTCAGAC"
## [100] "                                   ---------- ---------- ---------- ---------- ----------"
## [101] ""                                                                                         
## [102] "                                   ---------- ---------- ---------- ---------- ----------"
## [103] "                                   AAACACTGAG GAAAGCAGAC CTTTTGGAGG CTTACCTTCT ACTCCATCT-"
## [104] "                                   AGATACATAA AAAGAGTTAT CTTGTTAAAA GGAACAGTCC ATTTGAACTA"
## [105] "                                   ---------- ---------- ---------- ---------- ----------"
## [106] ""                                                                                         
## [107] "                                   ---------- ---------- ---------- ---------- ----------"
## [108] "                                   ---------- --GGGTCCTA AATTAAAAAT CTCGGCTTAG ACAAAGAACT"
## [109] "                                   GTCACATCTA AATTCTCCTT AATTGGGAAC TTAGCCTTT- ----------"
## [110] "                                   ---------- ---------- ---------- ---------- ----------"
## [111] ""                                                                                         
## [112] "                                   ---------"                                             
## [113] "                                   TTCACTACT"                                             
## [114] "                                   ---------"                                             
## [115] "                                   ---------"</code></pre>
</div>
<div id="session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] org.Mm.eg.db_3.13.0                      
##  [2] TxDb.Mmusculus.UCSC.mm10.knownGene_3.10.0
##  [3] org.Hs.eg.db_3.13.0                      
##  [4] TxDb.Hsapiens.UCSC.hg38.knownGene_3.13.0 
##  [5] GenomicFeatures_1.44.0                   
##  [6] AnnotationDbi_1.54.1                     
##  [7] Biobase_2.52.0                           
##  [8] BSgenome.Mmusculus.UCSC.mm10_1.4.0       
##  [9] BSgenome.Hsapiens.UCSC.hg38_1.4.3        
## [10] BSgenome.Drerio.UCSC.danRer10_1.4.2      
## [11] BSgenome_1.60.0                          
## [12] rtracklayer_1.52.0                       
## [13] Biostrings_2.60.1                        
## [14] XVector_0.32.0                           
## [15] GenomicRanges_1.44.0                     
## [16] GenomeInfoDb_1.28.1                      
## [17] IRanges_2.26.0                           
## [18] S4Vectors_0.30.0                         
## [19] BiocGenerics_0.38.0                      
## [20] BiocParallel_1.26.1                      
## [21] enhancerHomologSearch_0.99.0             
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7                matrixStats_0.59.0         
##  [3] fs_1.5.0                    bit64_4.0.5                
##  [5] filelock_1.0.2              progress_1.2.2             
##  [7] httr_1.4.2                  rprojroot_2.0.2            
##  [9] tools_4.1.0                 bslib_0.2.5.1              
## [11] utf8_1.2.1                  R6_2.5.0                   
## [13] DBI_1.1.1                   tidyselect_1.1.1           
## [15] prettyunits_1.1.1           bit_4.0.4                  
## [17] curl_4.3.2                  compiler_4.1.0             
## [19] textshaping_0.3.5           xml2_1.3.2                 
## [21] desc_1.3.0                  DelayedArray_0.18.0        
## [23] sass_0.4.0                  rappdirs_0.3.3             
## [25] pkgdown_1.6.1               systemfonts_1.0.2          
## [27] stringr_1.4.0               digest_0.6.27              
## [29] Rsamtools_2.8.0             rmarkdown_2.9              
## [31] pkgconfig_2.0.3             htmltools_0.5.1.1          
## [33] MatrixGenerics_1.4.0        dbplyr_2.1.1               
## [35] fastmap_1.1.0               rlang_0.4.11               
## [37] RSQLite_2.2.7               jquerylib_0.1.4            
## [39] BiocIO_1.2.0                generics_0.1.0             
## [41] jsonlite_1.7.2              dplyr_1.0.7                
## [43] RCurl_1.98-1.3              magrittr_2.0.1             
## [45] GenomeInfoDbData_1.2.6      Matrix_1.3-4               
## [47] Rcpp_1.0.7                  fansi_0.5.0                
## [49] lifecycle_1.0.0             stringi_1.6.2              
## [51] yaml_2.2.1                  SummarizedExperiment_1.22.0
## [53] zlibbioc_1.38.0             BiocFileCache_2.0.0        
## [55] grid_4.1.0                  blob_1.2.1                 
## [57] crayon_1.4.1                lattice_0.20-44            
## [59] hms_1.1.0                   KEGGREST_1.32.0            
## [61] msa_1.24.0                  knitr_1.33                 
## [63] pillar_1.6.1                rjson_0.2.20               
## [65] biomaRt_2.48.2              XML_3.99-0.6               
## [67] glue_1.4.2                  evaluate_0.14              
## [69] png_0.1-7                   vctrs_0.3.8                
## [71] purrr_0.3.4                 assertthat_0.2.1           
## [73] cachem_1.0.5                xfun_0.24                  
## [75] restfulr_0.0.13             ragg_1.1.3                 
## [77] tibble_3.1.2                GenomicAlignments_1.28.0   
## [79] memoise_2.0.0               ellipsis_0.3.2</code></pre>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-howe2013zebrafish" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Howe, K. <em>et al.</em> The zebrafish reference genome sequence and its relationship to the human genome. <em>Nature</em> <strong>496,</strong> 498–503 (2013).</div>
</div>
<div id="ref-braasch2016spotted" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Braasch, I. <em>et al.</em> The spotted gar genome illuminates vertebrate evolution and facilitates human-teleost comparisons. <em>Nature genetics</em> <strong>48,</strong> 427–437 (2016).</div>
</div>
<div id="ref-wong2020deep" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Wong, E. S. <em>et al.</em> Deep conservation of the enhancer regulatory code in animals. <em>Science</em> <strong>370,</strong> (2020).</div>
</div>
<div id="ref-pages2013package" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Pages, H., Aboyoun, P., Gentleman, R. &amp; DebRoy, S. Package <span>“biostrings”</span>. <em>R package version</em>
</div>
</div>
<div id="ref-bodenhofer2015msa" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">Bodenhofer, U., Bonatesta, E., Horejš-Kainrath, C. &amp; Hochreiter, S. Msa: An r package for multiple sequence alignment. <em>Bioinformatics</em> <strong>31,</strong> 3997–3999 (2015).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jianhong Ou, Kenneth Poss.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
